grpc-gcp/src/main/java/com/google/cloud/grpc/GcpMultiEndpointOptions.java

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.grpc;
+
+import com.google.api.core.ApiFunction;
+import com.google.cloud.grpc.multiendpoint.MultiEndpoint;
+import com.google.common.base.Preconditions;
+import io.grpc.ChannelCredentials;
+import io.grpc.ManagedChannelBuilder;
+import java.time.Duration;
+import java.util.List;
+
+/**
+ * {@link MultiEndpoint} configuration for the {@link GcpMultiEndpointChannel}.
+ */
+public class GcpMultiEndpointOptions {
+
+  private final String name;
+  private final List<String> endpoints;
+  private final ApiFunction<ManagedChannelBuilder<?>, ManagedChannelBuilder<?>> channelConfigurator;
+  private final ChannelCredentials channelCredentials;
```

Comment:
What is different endpoints in the same MultiEndpoints needs to use different channel credentials? The above example about DirectPath and CloudPath also applies here. This use case needs to be supported.

Reply:
Same as above.